File: Semestrul 4/MPP2/Problema8_SC_REST/Problema8SC_Rest/Problema8Networking/src/main/java/problema8/network/dto/DTOUtils.java
package problema8.network.dto;

import problema8.model.*;

public class DTOUtils {
    public static User getFromDTO(UserDTO userDTO) {
        String username = userDTO.getUsername();
        String password = userDTO.getPassword();
        return new User(username, password);
    }

    public static UserDTO getDTO(User user) {
        String username = user.getUsername();
        String password = user.getPassword();
        return new UserDTO(username, password);
    }

    public static UserDTO[] getDTO(User[] users) {
        UserDTO[] userDTOs = new UserDTO[users.length];
        for (int i = 0; i < users.length; i++) {
            userDTOs[i] = getDTO(users[i]);
        }
        return userDTOs;
    }

    public static User[] getFromDTO(UserDTO[] userDTOs) {
        User[] users = new User[userDTOs.length];
        for (int i = 0; i < userDTOs.length; i++) {
            users[i] = getFromDTO(userDTOs[i]);
        }
        return users;
    }

    public static AgeGroup getFromDTO(AgeGroupDTO ageGroupDTO) {
        return new AgeGroup(ageGroupDTO.getName(), ageGroupDTO.getMinAge(), ageGroupDTO.getMaxAge());
    }

    public static AgeGroupDTO getDTO(AgeGroup ageGroup) {
        return new AgeGroupDTO(ageGroup.getName(), ageGroup.getMinAge(), ageGroup.getMaxAge());
    }

    public static ChildDTO getDTO(Child child) {
        return new ChildDTO(child.getName(), child.getCNP());
    }

    public static EnrollmentDTO getDTO(Enrollment enrollment) {
        return new EnrollmentDTO(enrollment.getChild().getId(), enrollment.getEvent().getId());
    }

    public static EventDTO getDTO(Event event) {
        return new EventDTO(event.getName(), event.getDistance(), event.getAgeGroupId());
    }
}

File: Semestrul 4/MPP2/Problema8_Csharp_ORM/Networking/dto/DTOUtils.cs
namespace Networking.dto;

using Problema8SC_CSharp.Model;

public static class DTOUtils
{
    public static User GetFromDTO(UserDTO userDTO)
    {
        string username = userDTO.Username;
        string password = userDTO.Password;
        return new User(username, password);
    }

    public static UserDTO GetDTO(User user)
    {
        string username = user.Username;
        string password = user.Password;
        return new UserDTO(username, password);
    }

    public static UserDTO[] GetDTO(User[] users)
    {
        UserDTO[] userDTOs = new UserDTO[users.Length];
        for (int i = 0; i < users.Length; i++)
        {
            userDTOs[i] = GetDTO(users[i]);
        }
        return userDTOs;
    }

    public static User[] GetFromDTO(UserDTO[] userDTOs)
    {
        User[] users = new User[userDTOs.Length];
        for (int i = 0; i < userDTOs.Length; i++)
        {
            users[i] = GetFromDTO(userDTOs[i]);
        }
        return users;
    }

    public static AgeGroup GetFromDTO(AgeGroupDTO ageGroupDTO)
    {
        return new AgeGroup(ageGroupDTO.Name, ageGroupDTO.MinAge, ageGroupDTO.MaxAge);
    }

    public static AgeGroupDTO GetDTO(AgeGroup ageGroup)
    {
        return new AgeGroupDTO(ageGroup.Name, ageGroup.MinAge, ageGroup.MaxAge);
    }

    public static ChildDTO GetDTO(Child child)
    {
        return new ChildDTO(child.Name, child.CNP);
    }

    public static EnrollmentDTO GetDTO(Enrollment enrollment)
    {
        return new EnrollmentDTO(enrollment.Child.GetId(), enrollment.Event.GetId());
    }

    public static EventDTO GetDTO(Event eventObj)
    {
        return new EventDTO(eventObj.Name, eventObj.Distance, eventObj.AgeGroupId);
    }
}
